### PR TITLE
New UI for HomePage

### PR DESCRIPTION
**Pull request by xinyao-song**
Added a new scroll and UI for home
and basically fixed a bunch of issues and features with follow all and stuff 
check ALL features to see if it works

REMEMBER run all of these:
npm cache clean --force
rm -rf node_modules package-lock.json
npm install
npx react-native-asset

### CODE REVIEW

**Review by AllenHsm — APPROVED** [review]
Perfect! The search bar disappear when scrolling down!

File: GOLDTracker/src/components/SearchComponent.js [hunk 1]
```diff
@@ -1,24 +1,93 @@
...
+} from "react-native";
 import { SearchBar } from "react-native-elements";
 import Class from "../components/Class";
-import {useNavigation} from "@react-navigation/native";
+import { useNavigation } from "@react-navigation/native";
 import { auth } from "../../firebaseConfig";
-import CategorySearch from "../components/CategorySearch";
 import * as ClassRegister from "./ClassRegister";
+import { COLORS } from "../theme/theme";
+import departmentMapping from "../assets/departmentMapping.json";
```
Comment (AllenHsm) [inline GOLDTracker/src/components/SearchComponent.js]: should replace departmentMapping with departmentList.json

File: GOLDTracker/src/components/SearchComponent.js [hunk 2]
```diff
@@ -42,95 +111,172 @@ const SearchComponent = ({ search, setSearch, setIsSearching, major }) => {
...
         return;
       }
 
-      console.log("department code in search is: ", deptCode)
-      console.log("search.trim in search is: ", search.trim())
-      apiUrl = ""
-      if (deptCode){
-        apiUrl = /^[A-Z|a-z]{2,}\s[\d(A-Z|a-z]+$/.test(searchTerm)
-
+      const apiUrl = /^[A-Z]{2,}\s[\dA-Z]+$/.test(searchTerm)
```
Comment (AllenHsm) [inline GOLDTracker/src/components/SearchComponent.js]: For subject area search and department search, we should use two different parameters in API. You can see the difference in current main branch. 

      if (deptCode){
        apiUrl = /^[A-Z|a-z]{2,}\s[\d(A-Z|a-z]+$/.test(searchTerm)

        ? `${API_URL}?quarter=${quarter}&courseId=${encodeURIComponent(
            searchTerm
          )}&includeClassSections=true`
        : `${API_URL}?quarter=${quarter}&deptCode=${encodeURIComponent(
            searchTerm
          )}&includeClassSections=true&pageNumber=1&pageSize=30`;
      }
      else{
        apiUrl = /^[A-Z|a-z]{2,}\s[\d(A-Z|a-z]+$/.test(searchTerm)
        ? `${API_URL}?quarter=${quarter}&courseId=${encodeURIComponent(
            searchTerm
          )}&includeClassSections=true`
        : `${API_URL}?quarter=${quarter}&subjectCode=${encodeURIComponent(
            searchTerm
          )}&includeClassSections=true&pageNumber=1&pageSize=30`;
      }

File: GOLDTracker/src/components/SearchComponent.js [hunk 3]
```diff
@@ -1,24 +1,93 @@
...
+    inputRange: [0, searchBarHeight],
+    outputRange: [searchBarHeight + filterBarHeight, filterBarHeight],
+    extrapolate: "clamp",
+  });
+
+  const departmentOptions = [
+    { code: "Your Major", label: "Cancel Selection" },
+    ...Object.keys(departmentMapping).map((code) => ({
+      code,
+      label: departmentMapping[code][0],
```
Comment (AllenHsm) [inline GOLDTracker/src/components/SearchComponent.js]: remove the [0] here to fit the new mapping json. Should be: label: departmentMapping[code]

**Review by AllenHsm — CHANGES_REQUESTED** [review]
Need to correct several errors causing it unable to search subject areas like "CHIN" and the wrong department list.

**Review by AllenHsm — APPROVED** [review]
LGTM!!!

**Review by Simonmann17 — APPROVED** [review]
LGTM!

**Pull request merged**